env/build_cross_compile.bash

```diff
@@ -0,0 +1,271 @@
+
+    pretty_info "Building GCC with ${PROC_COUNT} threads"
+    runner "Failed to build GCC" make -j "${PROC_COUNT}" all-gcc
+    pretty_success "GCC built correctly"
+
+    pretty_info "Building libgcc"
+    runner "Failed to build libgcc" make -j "${PROC_COUNT}" all-target-libgcc
+    pretty_success "libgcc built correctly"
+
+    pretty_info "Installing GCC"
+    runner "Failed to install GCC" make install-gcc
+    pretty_success "GCC installed correctly"
+
+    pretty_info "Installing libgcc"
+    runner "Failed to install libgcc" make install-target-libgcc
+    pretty_success "libgcc installed correctly"
+
+    pretty_success "GCC build completed"
+}
+
+build_gdb() {
+  local gdb_dir="${CROSS_COMPILE_BUILD_BUILD_DIR}/build_gdb"
+  local gdb_name="gdb-${CROSS_COMPILE_BUILD_GDB_VER}"
+
+  pretty_info "Building GDB"
+
+  prepare_directory "${gdb_dir}"
+  cd "${gdb_dir}"
+
+  download_extract_gnu_source "${gdb_name}.tar.gz" "https://ftp.gnu.org/gnu/gdb/"
+
+  pretty_info "Configuring GDB"
+  runner "Failed to configure GDB" ${gdb_name}/configure --target=${CROSS_COMPILE_BUILD_TARGET} --prefix="${CROSS_COMPILE_BUILD_TOOL_DIR}" --disable-werror
+  pretty_success "GDB configured correctly"
+
+  pretty_info "Building GDB with ${PROC_COUNT} threads"
+  runner "Failed to build GDB" make -j "${PROC_COUNT}" all-gdb
+  pretty_success "GDB built correctly"
+
+  pretty_info "Installing GDB"
+  runner "Failed to install GDB" make install-gdb
+  pretty_success "GDB installed correctly"
+
+  pretty_success "GDB build completed"
+}
+
+check_path() {
+    local new_path="${CROSS_COMPILE_BUILD_TOOL_DIR}/bin"
+
+    pretty_info "Checking if ${new_path} is in PATH"
+
+    IFS=':' read -ra path_array <<< "$PATH"
+
+    local existing_path
+    for existing_path in "${path_array[@]}"; do
+        if [ "$existing_path" = "$new_path" ]; then
+            pretty_info "Path: ${new_path} already exists in PATH"
+            CROSS_COMPILE_BUILD_IS_ADDED_TO_PATH=true
+            return 0
+        fi
+    done
+
+    pretty_info "Path: ${new_path} does not exist in PATH"
+}
+
+add_to_user_path() {
+  local new_path="${CROSS_COMPILE_BUILD_TOOL_DIR}/bin"
+  pretty_info "Adding ${new_path} to PATH"
+
+  if [ "$CROSS_COMPILE_BUILD_IS_ADDED_TO_PATH" = false ]; then
+    echo "export PATH=\"${new_path}:\$PATH\"" >> ~/.bashrc
+    echo "export PATH=\"${new_path}:\$PATH\"" >> ~/.profile
+    pretty_success "Path: ${new_path} added to PATH"
+  else
+    pretty_info "Path: ${new_path} already exists in PATH"
+  fi
+}
+
+run_build() {
+  pretty_info "Starting GCC cross-compiler build with build directory: ${CROSS_COMPILE_BUILD_BUILD_DIR} and target directory: ${CROSS_COMPILE_BUILD_TOOL_DIR}"
+
+  check_path
+  export PATH="$CROSS_COMPILE_BUILD_TOOL_DIR/bin:$PATH"
+
+  build_binutils
+  build_gdb
+  build_gcc
+
+  add_to_user_path
+
+  pretty_success "Build: ${CROSS_COMPILE_BUILD_SCRIPT_PATH} completed successfully"
+}
+
+parse_args() {
+  while [[ $# -gt 0 ]]; do
```

Comment:
No option to parse quiet mode that already exists as argument to runner

Reply:
There was quiet argument for runner and verbose for CLI, fixed all of them now uses verbose